### PR TITLE
Fix ICE when pretty printing `println!{"hello world"}`

### LINKED ISSUE #26072: Pretty printer isn't preserving macro calls with brackets
This code block:

``` rust
fn main() {
    println!{"foo {}", "bar"}
}
```

While it compiles fine with `rustc test.rs`, when run with `rustc -Z unstable-options --pretty normal` it produces this, which doesn't compile:

``` rust
fn main() {
    println!("foo {}" , "bar")
}
```

### PR DESCRIPTION
**Pull request by erickt**
This fixes an ICE when pretty printing:

``` rust
fn main() {
    println!{"hello world"}
    ()
}
```

This also preserves braces in expression macros. Closes #26072.

### CODE REVIEW

File: src/test/pretty/issue-26072.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+//
+// Testing that unsafe blocks in match arms are followed by a comma
```
Comment (arielb1) [inline src/test/pretty/issue-26072.rs]: Copy-paste error?

Comment (erickt) [inline src/test/pretty/issue-26072.rs]: Thanks! I fixed it.

File: src/librustc/middle/check_const.rs [hunk 2]
```diff
@@ -770,7 +770,7 @@ fn check_expr<'a, 'tcx>(v: &mut CheckCrateVisitor<'a, 'tcx>,
         ast::ExprAssign(..) |
         ast::ExprAssignOp(..) |
         ast::ExprInlineAsm(_) |
-        ast::ExprMac(_) => {
+        ast::ExprMac(_, _) => {
```
Comment (nagisa) [inline src/librustc/middle/check_const.rs]: Why not `..` like in the two files below?

Comment (SOFTowaha) [inline src/librustc/middle/check_const.rs]: Looks good now :)

**Comment by alexcrichton** [comment]
Closing due to inactivity, but feel free to reopen with the tests fixed!

**Pull request closed without merging**